fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt

```diff
@@ -483,6 +485,26 @@ object TestHelper {
         }
     }
 
+    /**
+     * Wrapper to launch the app using the launcher intent.
+     */
+    fun runWithLauncherIntent(
+        activityTestRule: AndroidComposeTestRule<HomeActivityIntentTestRule, HomeActivity>,
+        testBlock: () -> Unit,
+    ) {
+        val launcherIntent = Intent(Intent.ACTION_MAIN).apply {
+            addCategory(Intent.CATEGORY_LAUNCHER)
+        }
+
+        activityTestRule.activityRule.withIntent(launcherIntent).launchActivity(launcherIntent)
+
+        try {
+            testBlock()
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
+    }
+
```

Comment:
Looking at `TestHelper.kt` currently:

I wouldn't mind breaking this out into a new `AppHelper.kt` as `TestHelper.kt` has so many miscellaneous functions. I think we can treat the new Class as app and system management helper methods and continue to use `TestHelper.kt` as UI interaction and verification functions (e.g, `verifyLightThemeApplied()`, `verifyDarkThemeApplied()`, and `runWithCondition()`)

App and system management functions e.g `getPermissionAllowID()`, `setNetworkEnabled()`, `isPackageInstalled()`, `assertExternalAppOpens()`, `assertNativeAppOpens()`, `assertYoutubeAppOpens()`, `isExternalAppBrowserActivityInCurrentTask()`, `registerAndCleanupIdlingResources()`, `grantSystemPermission()`, `denyPermission()`, `isTestLab()`, `runWithSystemLocaleChanged()`, `setSystemLocale()`, `putAppToBackground()`, `bringAppToForeground()`, `verifyKeyboardVisibility()`, `getRegionSearchEnginesList()`, and `getAvailableSearchEngines()`

There's also potential for a new data generation and manipulation Helper, `dataGenerationHelper.kt` Class as we have: `createCustomTabIntent(`, `createTestBitmap()`, `generateRandomString()`, `setTextToClipBoard()`, and `getSponsoredShortcutTitle()`.

Lastly we can keep `TestHelper.kt` for UI interaction and verification with functions like `scrollToElementByText()`, `longTapSelectItem()`, `restartApp()`, `closeApp()`, `relaunchCleanApp()`, `waitUntilObjectIsFound()`, `clickSnackbarButton()`, `waitUntilSnackbarGone()`, `verifySnackBarText()`, `verifyUrl()`, and `exitMenu()`.

Thoughts?